crates/oxc_linter/src/rules/unicorn/consistent_existence_index_check.rs

```diff
@@ -0,0 +1,493 @@
+        r"
+          	let index = foo.indexOf('bar');
+
+          	index < 0
+        ",
+        // To prevent false positives, it will not check if the index is not declared via const
+        r"
+          	var index = foo.indexOf('bar');
+          	index < 0
+        ",
+        // To prevent false positives, it will not check if the index is not declared via const
+        r"
+          	let index;
+
+          	// do stuff
+
+          	index = arr.findLastIndex(element => element > 10);
+          	index < 0;
+        ",
+        r"const indexOf = 'indexOf'; const index = foo[indexOf](foo); index < 0;",
+        r"const index = foo.indexOf?.(foo); index < 0;",
+        r"const index = foo?.indexOf(foo); index < 0;",
+    ];
+
+    let fail = vec![
+        r"const index = foo.indexOf('bar'); if (index < 0) {}",
+        r"const index = foo.lastIndexOf('bar'); if (index < 0) {}",
+        r"const index = foo.findIndex('bar'); if (index < 0) {}",
+        r"const index = foo.findLastIndex('bar'); if (index < 0) {}",
+        r"const index = foo.indexOf('bar'); if (index >= 0) {}",
+        r"const index = foo.lastIndexOf('bar'); if (index >= 0) {}",
+        r"const index = foo.findIndex('bar'); if (index >= 0) {}",
+        r"const index = foo.findLastIndex('bar'); if (index >= 0) {}",
+        r"const index = foo.indexOf('bar'); if (index > -1) {}",
+        r"const index = foo.lastIndexOf('bar'); if (index > -1) {}",
+        r"const index = foo.findIndex('bar'); if (index > -1) {}",
+        r"const index = foo.findLastIndex('bar'); if (index > -1) {}",
+        r"
+        	const index = foo.indexOf(bar);
+
+        	function foo () {
+        		if (index < 0) {}
+        	}
+        ",
+        r"
+        	const index1 = foo.indexOf('1'),
+        		index2 = foo.indexOf('2');
+        	index1 < 0;
+        	index2 >= 0;
+        ",
+        r"
+              	const index = foo.indexOf('1');
+              	((
+              		/* comment 1 */
+              		((
+              			/* comment 2 */
+              			index
+              			/* comment 3 */
+              		))
+              		/* comment 4 */
+              		<
+              		/* comment 5 */
+              		((
+              			/* comment 6 */
+              			0
+              			/* comment 7 */
+              		))
+              		/* comment 8 */
+              	));
+              ",
+        r"
+        	const index = foo.indexOf('1');
+        	((
+        		/* comment 1 */
+        		((
+        			/* comment 2 */
+        			index
+        			/* comment 3 */
+        		))
+        		/* comment 4 */
+        		>
+        		((
+        			/* comment 5 */
+        			- /* comment 6 */ (( /* comment 7 */ 1 /* comment 8 */ ))
+        			/* comment 9 */
+        		))
+        	));
+        ",
+        r"const index = _.indexOf([1, 2, 1, 2], 2); index < 0;",
+    ];
+
+    let fix = vec![
```

Comment:
Can we add all of the fail cases here as well, to make sure they have the correct output?